### PR TITLE
Add functional tests for newsletter page

### PR DESCRIPTION
**Pull request by schalkneethling**
@alexgibson r? Not 100% sure the way I am testing form submission 'fail' when validation fails but hey, that is what pull requests are for right ;)

### CODE REVIEW

**Comment by alexgibson** [comment]
Agree I'm not sure the current form failure test feels quite right. Perhaps it would be nice just to affirm that an error message gets displayed when the server returns an error. To bypass HTML5 form validation that browsers automatically apply, maybe we could use the CasperJS `evaluate` method to set a `novalidate` attribute in the form before submitting. Then we might just be able to test that the error list gets shown in response. Just an idea, haven't actually tried this yet - but something we could try and figure out.

**Comment by schalkneethling** [comment]
@alexgibson yeah, your suggestion sounds much better. I will take a look at this.

File: tests/functional/newsletter.js [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+        test.done();
+        helpers.done();
+    });
+
+});
+
+casper.test.begin('Newsletter, Validation failed, form submission prevented: ' + url, 1, function suite(test) {
+
+    casper.start(url).thenEvaluate(function() {
+        document.querySelector('#id_email').removeAttribute('required');
```
Comment (alexgibson) [inline tests/functional/newsletter.js]: Could this be simplified to `document.getElementById('newsletter-form').setAttribute('novalidate', true);`?

File: tests/functional/newsletter.js [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+        test.done();
+        helpers.done();
+    });
+
+});
+
+casper.test.begin('Newsletter, Click element labels: ' + url, 2, function suite(test) {
+
+    casper.start(url, function() {
+        this.fill('#newsletter-form', {
```
Comment (alexgibson) [inline tests/functional/newsletter.js]: Is this needed for the test?

File: tests/functional/newsletter.js [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+});
+
+casper.test.begin('Newsletter, Successful form submission: ' + url, 1, function suite(test) {
+
+    casper.start(url, function() {
+        this.fill('#newsletter-form', {
+            email: 'noreply@mozilla.com',
+            privacy: true
+        });
```
Comment (alexgibson) [inline tests/functional/newsletter.js]: If you pass `true` here at the end of `this.fill`, there's no need to fire the `click` below

**Comment by alexgibson** [comment]
Other than a couple of nits, this looks great! r+wc thanks @schalkneethling

File: tests/functional/newsletter.js (hunk 1 above)
Comment (schalkneethling) [inline tests/functional/newsletter.js]: Ha! I had no idea about that attribute :blush: . That is indeed much simpler.

File: tests/functional/newsletter.js (hunk 2 above)
Comment (schalkneethling) [inline tests/functional/newsletter.js]: As I am not actually submitting the form, nope, it is not.

**Comment by schalkneethling** [comment]
@alexgibson updated and squashed

**Pull request merged**